src/commands/user/export.ts

```diff
@@ -0,0 +1,75 @@
+import * as fs from 'fs'
+import * as _ from 'lodash'
+import path from 'path'
+
+import { flags } from '@oclif/command'
+import { Kommand } from '../../common'
+import { kuzzleFlags, } from '../../support/kuzzle'
+
+export default class UserExport extends Kommand {
+  static description = 'Exports users'
+
+  static flags = {
+    help: flags.help({}),
+    path: flags.string({
+      description: 'Dump directory',
+      default: 'users'
+    }),
+    exclude: flags.string({
+      description: 'Exclude users by matching their IDs',
+      default: '[]'
+    }),
+    ...kuzzleFlags,
+    protocol: flags.string({
+      description: 'Kuzzle protocol (http or websocket)',
+      default: 'ws',
+    }),
+  }
+
+  private excludes: string[] = []
+
+  async runSafe() {
+    const filename = path.join(this.flags.path, 'users.json')
+
+    this.logInfo(`Exporting users in ${filename} ...`)
+
+    fs.mkdirSync(this.flags.path, { recursive: true })
+
+    this.excludes = this.parseJs(this.flags.exclude)
+
+    const users = await this._dumpUsers()
+
+    const dump = {
+      type: 'users',
+      content: users
+    }
+
+    fs.writeFileSync(filename, JSON.stringify(dump, null, 2))
+
+    this.logOk(`${Object.keys(users).length} users dumped`)
+  }
+
+  async _dumpUsers() {
+    const users: any = {}
+
+    let results = await this.sdk?.security.searchUsers(
+      {},
+      { scroll: '5s', size: 100 })
```

Comment:
IMHO scroll duration and size should be configurable.

Reply:
Why? Kuzzle with automatically refresh the `scroll` between each call so those `5s` represent the maximal time to process the 100 users.
Even with a bigger batch, the only operation is to put the users in memory so `5s` is far than enough

About the size I will add the `batch-size` option in case people want to dump bigger or smaller batches.

This method may not work if you have something like 10K or 100K users since it keep all the users in memory before saving them. If we want to handle large amount of users we need to use stream like for the collection import/export but it will complexify the code a lot so IMHO we can stick with this for now

Reply:
I don't really like magic numbers in the code of an application or a library. I think you should make them configurable but keep the values you chose by default. :slightly_smiling_face:

Reply:
At least for the scroll size parameter :wink:

Reply:
Done :+1: And again about the scroll duration, IMHO it's not a magic number or whatever, I just need to put a duration to open a scroll session. In this use case there is no reason for the user to put a higher scroll duration and actually it might be dangerous to let users do so

Reply:
A bit less dangerous now that scroll contexts are automatically destroyed by Kuzzle once you fetch all its results ;)